File: java/eyes.selenium.java/src/main/java/com/applitools/eyes/selenium/TargetPathLocator.java
package com.applitools.eyes.selenium;

/**
 * target path locator
 */
public class TargetPathLocator {
  protected TargetPathLocator parent;
  protected PathNodeValue value;

  public TargetPathLocator() {
  }

  public TargetPathLocator(TargetPathLocator parent, PathNodeValue value) {
    this.parent = parent;
    this.value = value;
  }

  public PathNodeValue getValue() {
    return value;
  }

  public TargetPathLocator getParent() {
    return parent;
  }

}

File: dotnet/Eyes.Selenium4.DotNet/Fluent/TargetPathLocator.cs
namespace Applitools.Selenium
{
    public class TargetPathLocator
    {
        protected internal TargetPathLocator Parent { get; }
        protected internal  IPathNodeValue Value { get; }

        public TargetPathLocator()
        {
        }

        public TargetPathLocator(TargetPathLocator parent, IPathNodeValue value)
        {
            Parent = parent;
            Value = value;
        }
    }
}